### PR TITLE
fix: refactor list operations

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: momento/list_concatenate_front.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		Values:             values,
+		TtlMilliseconds:    ttl,
+		RefreshTtl:         r.CollectionTTL.RefreshTtl,
+		TruncateBackToSize: r.TruncateBackToSize,
+	}
+
+	return nil
+}
+
+func (r *ListConcatenateFrontRequest) makeGrpcRequest(client scsDataClient, ctx context.Context) (grpcResponse, error) {
```
Comment (eaddingtonwhite) [inline momento/list_concatenate_front.go:77]: ctx should always be first arg to func in golang. I think maybe govet should fail on this?

File: momento/list_remove_value.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	r.grpcRequest = &pb.XListRemoveRequest{
+		ListName: []byte(r.ListName),
+		Remove:   &pb.XListRemoveRequest_AllElementsWithValue{AllElementsWithValue: value},
+	}
+
+	return nil
+}
+
+func (r *ListRemoveValueRequest) makeGrpcRequest(client scsDataClient, ctx context.Context) (grpcResponse, error) {
```
Comment (eaddingtonwhite) [inline momento/list_remove_value.go:57]: same here should context be first?

File: momento/list_push_front.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		Value:              value,
+		TtlMilliseconds:    ttl,
+		RefreshTtl:         r.CollectionTTL.RefreshTtl,
+		TruncateBackToSize: r.TruncateBackToSize,
+	}
+
+	return nil
+}
+
+func (r *ListPushFrontRequest) makeGrpcRequest(client scsDataClient, ctx context.Context) (grpcResponse, error) {
```
Comment (eaddingtonwhite) [inline momento/list_push_front.go:78]: ctx first arg?

**Review by eaddingtonwhite — APPROVED** [review]
one nit on how were passing context in internal method not as first arg I think we should fix. I think that pattern was set before this PR though so im ok merging this we can iterate on future pr on that.

File: momento/list_concatenate_front.go (hunk 1 above)
Comment (pgautier404) [reply momento/list_concatenate_front.go:77]: Ok, yeah, this argument order is currently baked into all of the request classes. I'll fix it in a quick followup.

**Pull request merged**